test/test_snprintf.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include "rcutils/snprintf.h"
+
+// Tests the rcutils_snprintf() function.
+TEST(TestSnprintf, test_snprintf) {
+  const char * test_str = "0123456789";
+  char buffer[256];
+  int ret;
+
```

Comment:
be good to have a test for the `buffer=null, size=0` case, which we know has different behaviour in the underlying function on windows

Reply:
I'll add one.